A generic CONTRIBUTING file

gvwilson opened the pull request:
This adds a `CONTRIBUTING.md` file to `core` for use in all lessons based on one written by @ChristinaLK for the novice shell lesson.

gvwilson committed:
A generic CONTRIBUTING file

wking commented:
On Wed, Feb 11, 2015 at 06:18:14PM -0800, Greg Wilson wrote:

> - A generic CONTRIBUTING file
- There's some trailing whitespace in here.
- I prefer reference-style links [1](https://github.com/swcarpentry/shell-novice/pull/68#issuecomment-73185679).

I'm concerned about having both lesson-template-oriented content (“If
you're looking for things to work on, please see the current list of
issues”) and stuff setting up for downstream lessons (“This lesson is
based on the template…”).  Do we want to make that distinction more
obvious, so template contributors don't get confused and downstream
lesson maintainers know what to tweak?

rgaiacs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,42 @@
+---
+layout: page
```
Do we need this YAML header? I thought that `CONTRIBUTING.md` is like `README.md` or `FAQ.md`.

rgaiacs commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,42 @@
+
+By contributing,
+you are agreeing that Software Carpentry may redistribute your work under
+[these licenses](LICENSE.md).
```
This link will be "broken" at `CONTRIBUTING.html`.

wking replied:
On Thu, Feb 12, 2015 at 03:57:48AM -0800, Raniere Silva wrote:

> > +By contributing,
> > +you are agreeing that Software Carpentry may redistribute your work under
> > +[these licenses](LICENSE.md).
> 
> This link will be "broken" at `CONTRIBUTING.html`.

GitHub handles these links with it's file browser [1](https://github.com/blog/1395-relative-links-in-markup-files).  I don't know
if Pandoc does, but I don't think we want a CONTRIBUTING.html anyway.

rgaiacs replied:
I'm OK with this line be

```
[these licenses](LICENSE.md).
```

To clarify my first comment, if `CONTRIBUTING.html` was available the link should point to `LICENSE.html` that is already available because is probably the file that the user want. Since we are going to keep only `CONTRIBUTING.md` I'm +1 to link against `LICENSE.md`.

@wking Pandoc doesn't change the links:

```
$ pandoc -f markdown -t html <<EOF
[license](LICENSE.md)
EOF
<p><a href="LICENSE.md">license</a></p>
```

This is the same behavior of others Markdown to HTML:

```
$ kramdown <<EOF
[license](LICENSE.md)
EOF
<p><a href="LICENSE.md">license</a></p>
```

gvwilson committed:
Tidying up based on feedback:

1.  Removed YAML header.
2.  Changed link to refer to `LICENSE.html` instead of `LICENSE.md`.

Leaving the reference to the lesson-template repo alone - lesson authors will be expected to edit this (and instructions to do that will go in the docs in the gh-pages branch).

gvwilson committed:
Tidying up language a bit
